File: app/src/main/java/com/mikepenz/materialdrawer/app/utils/CrossfadeWrapper.java
package com.mikepenz.materialdrawer.app.utils;

import com.mikepenz.crossfader.Crossfader;
import com.mikepenz.materialdrawer.interfaces.ICrossfader;

/**
 * Created by mikepenz on 18.07.15.
 */
public class CrossfadeWrapper implements ICrossfader {
    private Crossfader mCrossfader;

    public CrossfadeWrapper(Crossfader crossfader) {
        this.mCrossfader = crossfader;
    }

    @Override
    public void crossfade() {
        mCrossfader.crossFade();
    }

    @Override
    public boolean isCrossfaded() {
        return mCrossfader.isCrossFaded();
    }
}

File: app/src/main/java/com/mikepenz/materialdrawer/app/utils/CrossfadeWrapper.kt
package com.mikepenz.materialdrawer.app.utils

import com.mikepenz.crossfader.Crossfader
import com.mikepenz.materialdrawer.interfaces.ICrossfader

/**
 * Created by mikepenz on 18.07.15.
 */
class CrossfadeWrapper(private val crossfader: Crossfader<*>) : ICrossfader {

    override val isCrossfaded: Boolean
        get() = crossfader.isCrossFaded()

    override fun crossfade() {
        crossfader.crossFade()
    }
}
